src/components/FormType.tsx

```diff
@@ -0,0 +1,123 @@
+import { useContext } from "react"
+import { ModalContext } from "~/state/ModalContext"
+
+type ModalStateSetter = (args_0: boolean) => void
+
+interface GitHubFormTypeProps {
+  modals: [ModalStateSetter, ModalStateSetter];
+}
+
+const FormType = () => {
+  const { eventTrigger, setRulesetModalOpened, setCreateModalOpened } = useContext(ModalContext);
+
+  return (
+    <div>
+      {eventTrigger === "mail" && <MailFormType />}
+      {eventTrigger === "calendar" && <CalendarFormType />}
+      {eventTrigger === "github" && <GitHubFormType modals={[setCreateModalOpened, setRulesetModalOpened]} />}
+    </div>
+  );
+};
+
+
+const MailFormType = () => (
+  <div>
+    this is a mail only form
+  </div>
+)
+
+const CalendarFormType = () => (
+  <div>
+    this is a calendar only form
+  </div>
+)
+const GitHubFormType = ({ modals }: GitHubFormTypeProps) => {
+
+  const [C, RS] = modals
+  
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault()
+    // create automation and webhook logic func calls here
+    // show some UI if worked for failed
+
+    // close modals
+    C(false)
+    RS(false)
+  }
+
+  return (
+    <form className="space-y-4" onSubmit={handleSubmit}>
+        <div>
+          <label htmlFor="eventType" className="block font-medium">
+            Type of Event:
+          </label>
+          <select
+            id="eventType"
+            name="eventType"
+            className="border border-gray-300 rounded px-3 py-2 w-full"
+          >
+            <option value="" disabled>
+              Select event trigger here...
+            </option>
+            <option value="issueCreated">Issue Created in Repository</option>
```

Comment:
for future reference once we end up with a lot of rules/conditions, we might want to think about defining the per-option content somewhere else so that we don't have to repeat/hardcode the styles and elements

Reply:
Sounds good. My goal is still to have a working e2e implementation of one use case. Once I have this effectively working, I'll make the appropriate changes for rules, conditions, styling, and anything else effected.